crates/router/src/core/blocklist/utils.rs

```diff
@@ -598,68 +463,28 @@ pub async fn generate_payment_fingerprint(
-        .and_then(|pm_data| match pm_data {
-            api_models::payments::PaymentMethodData::Card(card) => {
-                crypto::HmacSha512::sign_message(
-                    &crypto::HmacSha512,
-                    merchant_fingerprint_secret.as_bytes(),
-                    card.card_number.clone().get_card_no().as_bytes(),
-                )
-                .attach_printable("error in pm fingerprint creation")
-                .map_or_else(
-                    |err| {
-                        logger::error!(error=?err);
-                        None
-                    },
-                    Some,
-                )
-            }
-            _ => None,
-        })
-        .map(hex::encode);
 
-    let mut fingerprint_id = None;
-    if let Some(encoded_hash) = card_number_fingerprint {
-        #[cfg(feature = "kms")]
-        let encrypted_fingerprint = kms::get_kms_client(&state.conf.kms)
-            .await
-            .encrypt(encoded_hash)
-            .await
-            .map_or_else(
-                |e| {
-                    logger::error!(error=?e, "failed kms encryption of card fingerprint");
-                    None
-                },
-                Some,
-            );
-
-        #[cfg(not(feature = "kms"))]
-        let encrypted_fingerprint = Some(encoded_hash);
-
-        if let Some(encrypted_fingerprint) = encrypted_fingerprint {
-            fingerprint_id = db
-                .insert_blocklist_fingerprint_entry(
-                    diesel_models::blocklist_fingerprint::BlocklistFingerprintNew {
-                        merchant_id,
-                        fingerprint_id: utils::generate_id(consts::ID_LENGTH, "fingerprint"),
-                        encrypted_fingerprint,
-                        data_kind: common_enums::BlocklistDataKind::PaymentMethod,
-                        created_at: common_utils::date_time::now(),
-                    },
-                )
-                .await
-                .map_or_else(
-                    |e| {
-                        logger::error!(error=?e, "failed storing card fingerprint in db");
-                        None
-                    },
-                    |fp| Some(fp.fingerprint_id),
-                );
+    if let Some(api_models::payments::PaymentMethodData::Card(card)) = payment_method_data.as_ref()
+    {
+        if let Some(payload) = generate_fingerprint(
+            state,
+            card.card_number.clone().get_card_no(),
+            merchant_fingerprint_secret,
+            api_models::enums::LockerChoice::Tartarus,
+        )
+        .await
+        .attach_printable("error in pm fingerprint creation")
+        .map_or_else(
+            |err| {
+                logger::error!(error=?err);
+                None
+            },
+            Some,
+        ) {
+            return Ok(Some(payload.card_fingerprint));
         }
     }
-    Ok(fingerprint_id)
+    logger::error!("failed to retrieve card fingerprint");
+    Ok(None)
```

Comment:
Same here, the `if ... else ...` is adding unnecessary visual complexity on of the code if flowing.